src/core/directives/google-map-polyline.ts

```diff
@@ -0,0 +1,224 @@
+
+  constructor(private _polylineManager: PolylineManager) { this._id = (polylineId++).toString(); }
+
+  /** @internal */
+  ngAfterContentInit() {
+    if (this._points.length) {
+      this._points.forEach((point: SebmGoogleMapPolylinePoint) => {
+        point.positionChanged.subscribe(
+            () => { this._polylineManager.updatePolylinePoints(this); });
+      });
+    }
+    if (!this._polylineAddedToManager) {
+      this._init();
+    }
+    // FIXME: once QueryList<> implements the Observable interface we can do this as needed
+    // FIXME: we are also limited to a fixed set of path items so far
+    this._polylineManager.updatePolylinePoints(this);
+  }
+
+  ngOnChanges(changes: SimpleChanges): any {
+    if (!this._polylineAddedToManager) {
+      this._init();
+      return;
+    }
+
+    if (changes['draggable']) {
+      this._polylineManager.updateDraggable(this);
+    }
+    if (changes['editable']) {
+      this._polylineManager.updateEditable(this);
+    }
+    if (changes['visible']) {
+      this._polylineManager.updateVisible(this);
+    }
+    if (changes['geodesic']) {
+      this._polylineManager.updateGeodesic(this);
+    }
+    if (changes['strokeColor']) {
+      this._polylineManager.updateStrokeColor(this);
+    }
+    if (changes['strokeOpacity']) {
+      this._polylineManager.updateStrokeOpacity(this);
+    }
+    if (changes['strokeWeight']) {
+      this._polylineManager.updateStrokeWeight(this);
+    }
+    if (changes['zIndex']) {
+      this._polylineManager.updateZIndex(this);
+    }
+  }
+
+  private _init() {
+    this._polylineManager.addPolyline(this);
+    this._polylineAddedToManager = true;
+    this._addEventListeners();
+  }
+
+  private _addEventListeners() {
+    const handlers = [
+      {name: 'click', handler: (ev: PolyMouseEvent) => this.lineClick.emit(ev)},
+      {name: 'dbclick', handler: (ev: PolyMouseEvent) => this.lineDblClick.emit(ev)},
+      {name: 'drag', handler: (ev: MouseEvent) => this.lineDrag.emit(ev)},
+      {name: 'dragend', handler: (ev: MouseEvent) => this.lineDragEnd.emit(ev)},
+      {name: 'dragstart', handler: (ev: MouseEvent) => this.lineDragStart.emit(ev)},
+      {name: 'mousedown', handler: (ev: PolyMouseEvent) => this.lineMouseDown.emit(ev)},
+      {name: 'mousemove', handler: (ev: PolyMouseEvent) => this.lineMouseMove.emit(ev)},
+      {name: 'mouseout', handler: (ev: PolyMouseEvent) => this.lineMouseOut.emit(ev)},
+      {name: 'mouseover', handler: (ev: PolyMouseEvent) => this.lineMouseOver.emit(ev)},
+      {name: 'mouseup', handler: (ev: PolyMouseEvent) => this.lineMouseUp.emit(ev)},
+      {name: 'rightclick', handler: (ev: PolyMouseEvent) => this.lineRightClick.emit(ev)},
+    ];
+    handlers.forEach((obj) => {
+      const os = this._polylineManager.createEventObservable(obj.name, this).subscribe(obj.handler);
+      this._observableSubscriptions.push(os);
+    });
+  }
+
+  getPoints(): Array<SebmGoogleMapPolylinePoint> {
+    if (this._points) {
```

Comment:
This won't work. QueryList doesn't implement the `Array<T>` interface which makes typescript unhappy (rightful). So we've to use the `.toArray()` method on it. Since `_points` is undefined during the first ngChanges (which trigger initialization of the underlaying google maps object) we've to provide a sane default (empty array).

If it is purely about the code style we could always resort to `this._points ? this._points.toArray() : []`. But that is not readable IMHO.